File: JS/qr-generator.js
const QRCode = require('qrcode');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const askQuestion = (query) => new Promise(resolve => rl.question(query, resolve));

async function createQRImage(data, filename) {
    try {
        await QRCode.toFile(filename, data, {
            errorCorrectionLevel: 'H',
            margin: 4,
            width: 300,
            color: { dark: '#000000', light: '#FFFFFF' }
        });
        console.log(`\nSuccess! QR code saved successfully as '${filename}'.\n`);
    } catch (err) {
        console.error(`\nError generating QR code:`, err, `\n`);
    }
}

async function menu() {
    while (true) {
        console.log("=== Permanent QR Code Generator ===");
        console.log("1. Webpage URL");
        console.log("2. WiFi Login");
        console.log("3. Exit");
        
        const choice = (await askQuestion("Select an option (1-3): ")).trim();

        if (choice === '1') {
            const url = (await askQuestion("Enter the webpage URL (e.g., https://example.com): ")).trim();
            if (!url) {
                console.log("\nURL cannot be empty.\n");
                continue;
            }
            
            let filename = (await askQuestion("Enter filename to save as (default: url_qr.png): ")).trim();
            if (!filename) filename = "url_qr.png";
            
            await createQRImage(url, filename);

        } else if (choice === '2') {
            const ssid = (await askQuestion("Enter the WiFi Network Name (SSID): ")).trim();
            if (!ssid) {
                console.log("\nNetwork Name (SSID) cannot be empty.\n");
                continue;
            }

            const password = (await askQuestion("Enter the WiFi Password (leave blank if none): ")).trim();
            
            console.log("Select Encryption Type:");
            console.log("1. WPA/WPA2/WPA3 (Most common)");
            console.log("2. WEP (Older, less secure)");
            console.log("3. None (Open network)");
            const encChoice = (await askQuestion("Choose (1-3, default 1): ")).trim();
            
            let encryption = 'WPA';
            if (encChoice === '2') encryption = 'WEP';
            if (encChoice === '3' || !password) encryption = 'nopass';

            const hiddenChoice = (await askQuestion("Is this a hidden network? (y/n, default n): ")).trim().toLowerCase();
            const hiddenStr = hiddenChoice === 'y' ? 'true' : 'false';

            const wifiData = `WIFI:T:${encryption};S:${ssid};P:${password};H:${hiddenStr};;`;

            let filename = (await askQuestion("Enter filename to save as (default: wifi_qr.png): ")).trim();
            if (!filename) filename = "wifi_qr.png";

            await createQRImage(wifiData, filename);

        } else if (choice === '3') {
            console.log("Exiting.");
            rl.close();
            break;
        } else {
            console.log("\nInvalid choice. Please enter 1, 2, or 3.\n");
        }
    }
}

menu();

File: python/qr-generator.py
import qrcode
import sys

def create_qr_image(data, filename):
    try:
        qr = qrcode.QRCode(
            version=1,
            error_correction=qrcode.constants.ERROR_CORRECT_H,
            box_size=10,
            border=4,
        )
        qr.add_data(data)
        qr.make(fit=True)

        img = qr.make_image(fill_color="black", back_color="white")
        img.save(filename)
        print(f"\nSuccess! QR code saved successfully as '{filename}'.\n")
    except Exception as e:
        print(f"\nError generating QR code: {e}\n")

def menu():
    while True:
        print("=== Permanent QR Code Generator ===")
        print("1. Webpage URL")
        print("2. WiFi Login")
        print("3. Exit")
        
        choice = input("Select an option (1-3): ").strip()

        if choice == '1':
            url = input("Enter the webpage URL (e.g., https://example.com): ").strip()
            if not url:
                print("\nURL cannot be empty.\n")
                continue
            
            filename = input("Enter filename to save as (default: url_qr.png): ").strip() or "url_qr.png"
            create_qr_image(url, filename)

        elif choice == '2':
            ssid = input("Enter the WiFi Network Name (SSID): ").strip()
            if not ssid:
                print("\nNetwork Name (SSID) cannot be empty.\n")
                continue
                
            password = input("Enter the WiFi Password (leave blank if none): ").strip()
            
            print("Select Encryption Type:")
            print("1. WPA/WPA2/WPA3 (Most common)")
            print("2. WEP (Older, less secure)")
            print("3. None (Open network)")
            enc_choice = input("Choose (1-3, default 1): ").strip()
            
            encryption = "WPA"
            if enc_choice == '2':
                encryption = "WEP"
            elif enc_choice == '3' or not password:
                encryption = "nopass"

            hidden_choice = input("Is this a hidden network? (y/n, default n): ").strip().lower()
            hidden_str = "true" if hidden_choice == 'y' else "false"

            wifi_data = f"WIFI:T:{encryption};S:{ssid};P:{password};H:{hidden_str};;"
            
            filename = input("Enter filename to save as (default: wifi_qr.png): ").strip() or "wifi_qr.png"
            create_qr_image(wifi_data, filename)

        elif choice == '3':
            print("Exiting.")
            sys.exit(0)
            
        else:
            print("\nInvalid choice. Please enter 1, 2, or 3.\n")

if __name__ == "__main__":
    menu()